modules/generator/storage/config_util.go

```diff
@@ -13,43 +13,51 @@ import (
 
 // generateTenantRemoteWriteConfigs creates a copy of the remote write configurations with the
 // X-Scope-OrgID header present for the given tenant, unless Tempo is run in single tenant mode or instructed not to add X-Scope-OrgID header.
-func generateTenantRemoteWriteConfigs(originalCfgs []prometheus_config.RemoteWriteConfig, tenant string, headers map[string]string, addOrgIDHeader bool, logger log.Logger, sendNativeHistograms bool) []*prometheus_config.RemoteWriteConfig {
-	var cloneCfgs []*prometheus_config.RemoteWriteConfig
+func generateTenantRemoteWriteConfigs(inputs []prometheus_config.RemoteWriteConfig, tenant string, headers map[string]string, addOrgIDHeader bool, logger log.Logger, sendNativeHistograms bool) []*prometheus_config.RemoteWriteConfig {
+	var outputs []*prometheus_config.RemoteWriteConfig
 
-	for _, originalCfg := range originalCfgs {
-		cloneCfg := &prometheus_config.RemoteWriteConfig{}
-		*cloneCfg = originalCfg
+	for _, input := range inputs {
+		output := &prometheus_config.RemoteWriteConfig{}
+		*output = input
+
+		// Copy headers so we can modify them
+		output.Headers = copyMap(output.Headers)
+
+		// Inject/overwrite custom headers from runtime overrides
+		for k, v := range headers {
+			output.Headers[k] = v
+		}
 
 		// Inject/overwrite X-Scope-OrgID header in multi-tenant setups
```

Comment:
Maybe we should do a slight update since the logic changed
```suggestion
		// Inject X-Scope-OrgID header in multi-tenant setups if not set already
```